File: python/readworkspace.py

# a python code to open the workspace

# import things from ROOT
from ROOT import TFile, TCanvas, RooWorkspace

# load this project's library
from utils import load_shared_object
load_shared_object()
from ROOT import RooExpAndGauss

def main():
    # open the file
    wsfile = TFile("workspace.root", "READ")
    ws = wsfile.Get("ws")
    ws.Print()

    # get things from the workspace
    x = ws.var("x")
    pdf = ws.pdf("pdf")

    # plot it again
    canvas = TCanvas()
    plot = x.frame()
    pdf.plotOn(plot)
    plot.Draw()
    canvas.SaveAs("plots/readFromWorkspaceExpAndGauss.pdf")

    return

# run the main function
if __name__ == "__main__":
    main()

File: main/readworkspace.cpp

// ROOT
#include "TCanvas.h"
#include "TFile.h"

// RooFit
#include "RooRealVar.h"
#include "RooDataSet.h"
#include "RooArgSet.h"
#include "RooPlot.h"
#include "RooWorkspace.h"

// from this framework
#include "RooExpAndGauss.hpp"

int main()
{
  // open the workspace
  TFile wsfile("workspace.root", "READ");
  RooWorkspace *ws = 0;
  wsfile.GetObject("ws", ws);
  ws->Print(); // hello!

  // get the pdf 
  RooExpAndGauss* pdf = (RooExpAndGauss*)ws->pdf("pdf");
  RooRealVar* x = ws->var("x");
 
  // plot it
  TCanvas canvas;
  RooPlot *plot = x->frame(50);
  pdf->plotOn(plot);
  plot->Draw();
  canvas.SaveAs("plots/afterSavingWorkspaceExpAndGauss.pdf");

  // tidy up
  wsfile.Close();
  delete plot;
  return 0;
}
